marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/step/saveStep.sjs

```diff
@@ -35,7 +35,15 @@ let existingStep = fn.head(cts.search(cts.andQuery([
 
 if (existingStep) {
   xdmp.trace(consts.TRACE_STEP, `Step with name ${stepName} and type ${stepDefinitionType} already exists, so will update`);
-  let updatedStep = Object.assign(existingStep.toObject(), stepProperties);
+  const isIngestionStep = "ingestion" === stepDefinitionType.toLowerCase() ? true : false;
```

Comment:
Can we do this in the loadData.sjs module instead? This endpoint should be as generic as possible; ideally, the step-type-specific logic is all in modules specific to a step type. 

Looks like validateArtifact would be a simple way to do it.